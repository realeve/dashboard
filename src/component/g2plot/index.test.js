import React from 'react';
import G2Plot from '@/component/g2plot';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const config = {
  chartType: 'bar',
  autoFit: true,
  data: [
    {
      type: '家具家电',
      sales: 38,
    },
    {
      type: '粮油副食',
      sales: 52,
    },
    {
      type: '生鲜水果',
      sales: 61,
    },
  ],
  xField: 'type',
  yField: 'sales',
};

// umi test ./src/component/g2plot/index.test.js
describe('g2plot', () => {
  test('初始化以及销毁', async () => {
    const { container, rerender } = render(<G2Plot renderer="svg" option={config} />);
    expect(container.innerHTML).toMatchSnapshot();

    rerender(<G2Plot renderer="svg" option={config} loading />);

    expect(container.querySelector('title')).toHaveTextContent('Loading...');

    const { chartType, ...option } = config;
    rerender(<G2Plot renderer="svg" option={option} />);

    expect(container.innerHTML).toMatchSnapshot();

    rerender(<G2Plot renderer="svg" option={{ chartType: 'unknown', ...option }} />);

    expect(container.innerHTML).toMatchSnapshot();
  });
});
