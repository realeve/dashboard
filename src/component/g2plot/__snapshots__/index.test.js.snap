// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`g2plot 初始化以及销毁 1`] = `"<div class=\\"\\" style=\\"height: 100%;\\" data-chart-source-type=\\"G2Plot\\"><div style=\\"position:relative;\\"><svg width=\\"400\\" height=\\"400\\" style=\\"width: 400px; height: 400px; display: inline-block; vertical-align: middle;\\"><defs id=\\"defs_1\\"></defs><g opacity=\\"1\\"><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g></g></svg></div></div>"`;

exports[`g2plot 初始化以及销毁 2`] = `"<div class=\\"\\" style=\\"height: 100%;\\" data-chart-source-type=\\"G2Plot\\"><div style=\\"position:relative;\\"><svg width=\\"400\\" height=\\"400\\" style=\\"width: 400px; height: 400px; display: inline-block; vertical-align: middle;\\"><defs id=\\"defs_1\\"></defs><g opacity=\\"1\\"><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g></g></svg></div><div style=\\"position:relative;\\"><svg width=\\"400\\" height=\\"400\\" style=\\"width: 400px; height: 400px; display: inline-block; vertical-align: middle;\\"><defs id=\\"defs_2\\"></defs><g opacity=\\"1\\"><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g><g opacity=\\"1\\"></g></g></svg></div></div>"`;

exports[`g2plot 初始化以及销毁 3`] = `"<h5>图表类型无效</h5>"`;
